packages/outputarea/src/model.ts

```diff
@@ -309,24 +309,21 @@ export class OutputAreaModel implements IOutputAreaModel {
     // Consolidate outputs if they are stream outputs of the same kind.
     if (
       nbformat.isStream(value) &&
-      this._lastStream &&
-      value.name === this._lastName &&
+      value.name === this._lastStreamName &&
       this.shouldCombine({
         value,
         lastModel: this.list.get(this.length - 1)
       })
     ) {
-      // In order to get a list change event, we add the previous
-      // text to the current item and replace the previous item.
-      // This also replaces the metadata of the last item.
-      this._lastStream += value.text as string;
-      this._lastStream = Private.removeOverwrittenChars(this._lastStream);
-      value.text = this._lastStream;
-      const item = this._createItem({ value, trusted });
+      // We append the new text to the current text list.
+      // This creates a text list change event.
       const index = this.length - 1;
-      const prev = this.list.get(index);
-      this.list.set(index, item);
-      prev.dispose();
+      const prev = this.list.get(index) as IOutputModel;
+      const textList = prev.observableData.get(
+        'text'
+      ) as unknown as IObservableList<string>;
+      const newText = Private.removeOverwrittenChars(value.text as string);
```

Comment:
Here `removeOverwrittenChars` is only called on the new text, not on the whole concatenated text as previously. This means that a backspace in a stream output data cannot affect the previous data. For instance:
```python
from time import sleep

print("abc", end="")
sleep(0.5)
print("\n")
```
Will not not result in the output `ab`, but in `abc`.